remotedata/src/main/java/com/mercari/remotedata/RemoteData.kt

```diff
@@ -15,14 +15,26 @@ sealed class RemoteData<out V : Any, out E : Exception> {
 
     object NotAsked : RemoteData<Nothing, Nothing>()
 
-    class Loading<V : Any> : RemoteData<V, Nothing>() {
+    class Loading<V : Any>(percentage: Int? = null) : RemoteData<V, Nothing>() {
+
+        var percentage: Int? = percentage
```

Comment:
Can you try `Delegates.vetoable`? 
```kotlin
var percentage by Delegates.vetoable(0) {
  //check condition inside?
}
```

And I am not a big fan of throwing though. I think if you set something more than 100, I would say it merge into 100 `value.coerceAtMost(100)` `value.coerceAtLeast(0)` otherwise.

Reply:
Ah sure that works, maybe we can also check if the new value is larger than the old one too.

Reply:
The tradeoff with `vetoable` is that we will need an additional backing variable.
It could nicer without, but I changed the Int to a sealed class to avoid the null case and make the type more explicit. Concerning exceptions, I'm a bit on the fence on that one since they do help to spot issues; but if you feel strongly against it, ignoring or coercing would be simple to do.